publisher-sdk/src/androidTest/java/com/criteo/publisher/advancednative/CriteoImageLoaderTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.advancednative;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.widget.ImageView;
+import com.criteo.publisher.DependencyProvider;
+import com.criteo.publisher.concurrent.ThreadingUtil;
+import com.criteo.publisher.concurrent.TrackingCommandsExecutor;
+import com.criteo.publisher.tests.R;
+import com.criteo.publisher.mock.MockedDependenciesRule;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.inject.Inject;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class CriteoImageLoaderTest {
+
+  @Rule
+  public MockedDependenciesRule mockedDependenciesRule = new MockedDependenciesRule();
+
+  @Inject
+  private Context context;
+
+  @Test
+  public void givenKittenUrl_ImageLoader_LoadsIt() throws Exception {
+    ThreadingUtil.runOnMainThreadAndWait(() -> {
+      LayoutInflater inflater = LayoutInflater.from(context);
+      ImageView imageView = (ImageView) inflater.inflate(R.layout.test_image_loader, null);
+
+      DependencyProvider dependencyProvider = mockedDependenciesRule.getDependencyProvider();
+      CriteoImageLoader imageLoader = (CriteoImageLoader) dependencyProvider
```

Comment:
I understand what you are saying, but I have a different point of view. I am not testing the image loader here. I am testing that picasso works well through `CriteoImageLoader` which we know depends on `Picasso`. Testing against an `ImageLoader` doesn't guarantee that `CriteoImageLoader` will be injected. Someone might mock the image loader and set that mock in the `MockedDependencyRule` for instance. OTOH, sure it can also happen that someone mocks `CriteoImageLoader`... I am fine either way, just wanted to explain why I used a `CriteoImageLoader` directly.